File: CSharp/knapsack.py
def knapsack(items, values, capacity, count):

    k = [[ 0 for i in range(capacity+1)] for j in range(count+1)]
    for i in range(count):
        for j in range(capacity):
            if i==0 and j == 0:k[i][j] = 0
            elif  values[i-1] <= j:
                k[i][j] = max(values[i - 1] + k[i - 1][j - values[i - 1]], k[i - 1][j])
            else:
                k[i][j] = k[i - 1][j];
    return k[count][capacity]


value = [10, 50, 70 ]
weight = [ 10, 20, 30 ]
capacity = 40;
itemsCount = 3;
result = knapsack(weight,value,capacity,itemsCount);
print(result)

File: CSharp/knapsack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Diagnostics;

namespace KnapsackAlgo
{
    class KnapsackAlgorithm
    {

        public static int KnapSack(int capacity, int[] weight, int[] value, int itemsCount)
        {
            int[,] K = new int[itemsCount + 1, capacity + 1];

            for (int i = 0; i <= itemsCount; ++i)
            {
                for (int w = 0; w <= capacity; ++w)
                {
                    if (i == 0 || w == 0)
                        K[i, w] = 0;
                    else if (weight[i - 1] <= w)
                        K[i, w] = Math.Max(value[i - 1] + K[i - 1, w - weight[i - 1]], K[i - 1, w]);
                    else
                        K[i, w] = K[i - 1, w];
                }
            }

            return K[itemsCount, capacity];
        }
        static void Main(string[] args)
        {
            int[] value = { 10, 50, 70 };
            int[] weight = { 10, 20, 30 };
            int capacity = 40;
            int itemsCount = 3;

            int result = KnapSack(capacity, weight, value, itemsCount);
            Console.WriteLine(result);
        }
    }
}